apps/transmitmedia.cpp

```diff
@@ -998,8 +1000,16 @@ class UdpSource: public Source, public UdpCommon
     UdpSource(string host, int port, const map<string,string>& attr)
     {
         Setup(host, port, attr);
-        int stat = ::bind(m_sock, sadr.get(), sadr.size());
-        if ( stat == -1 )
+
+        // In most circumstances, bind to the target addr (for unicast this is usually local)
+        // In linux multicast, bind to the remote address (224.0.0.0/4)
+        // In Windows multicast, bind to the local address
+        int stat =
+#if defined(_WIN32) || defined(__CYGWIN__)
+            is_multicast ? ::bind(m_sock, interface_addr.get(), interface_addr.size()) :
+#endif
+            ::bind(m_sock, target_addr.get(), target_addr.size());
```

Comment:
It is better without cpp entanglement.
```suggestion
#if defined(_WIN32) || defined(__CYGWIN__)
            sockaddr_any bindaddr = is_multicast ? interface_addr : target_addr;
#else
            sockaddr_any bindaddr = target_addr;
#endif 
            ::bind(m_sock, bindaddr.get(), bindaddr.size());
```